mvlearn/utils/utils.py

```diff
@@ -82,7 +82,11 @@ def check_Xs(
             )
             raise ValueError(msg)
 
-    Xs = [check_array(X, allow_nd=False, copy=copy) for X in Xs]
+    if type(Xs[0]) != np.memmap:
+        Xs = [check_array(X, allow_nd=False, copy=copy) for X in Xs]
+    else:
+        for X in Xs:
+            check_array(X)
```

Comment:
not clear what you want to do here. Avoid a copy when memmap? why ignoring the copy param for memmap I find this behavior hard to document. Can we update sklearn check_array to allow memmap and enforce here that copy=False when it's a memmap?

Reply:
The problem I'd like to avoid is that when you do check_array with a memmap, it returns an array (and not a memmap even when copy=False).
But you are right it would be better to do a PR in sklearn so that it supports memmap. I'll close this then.